c/include/proton/delivery.h

```diff
@@ -287,6 +287,12 @@ PN_EXTERN void pn_delivery_abort(pn_delivery_t *delivery);
  * @note If pn_delivery_current(delivery) is true before the call then
  * pn_link_advance(pn_delivery_link(deliver)) is called automatically.
  *
+ * @note The sender **should not** settle after only receiving a terminal
+ * status disposition with no settle flag, as then there would then be no way
+ * to receive any further events for that delivery (such as the subsequent
+ * on_settle message that might be expected when the receiver finally settles
+ * the message).
+ *
```

Comment:
What does settlement have to do with delivery abort? If there is a connection it needs to be called out explicitly to explain why this comment is here. Afaict If the sender aborts a delivery then it's likely they actually *do* want to settle the delivery because the delivery is now effectively dead and why would they want any more events about it? If you are making a comment relevant to this point then say that explicitly.
Essentially I think **should not** is way too strong here - if the application has no need for the receive settlement after aborting the message, and it seems to me likely they don't, then settling the delivery at the sender end first should be fine.

Reply:
I deleted this note.